### PR TITLE
Adding Simulate slit 

### PR DESCRIPTION
**Pull request by arunavabasu-03**
Add slit as a field so that users may apply it as they want !!

### CODE REVIEW

File: server/main.py [hunk 1]
```diff
@@ -59,14 +62,17 @@ async def calculate_spectrum(payload: Payload):
             databank=payload.database,
             use_cached=True,
         )
+        if simulate_slit_value == True:
+            spectrum.apply_slit(payload.simulate_slit, "nm")
+        else:
+            pass
```
Comment (suzil) [inline server/main.py:68]: I think we can simplify it a bit to just check if the value is None or not.
```suggestion
        if simulate_slit is not None:
            spectrum.apply_slit(payload.simulate_slit, "nm")
```

File: server/main.py [hunk 2]
```diff
@@ -59,14 +59,17 @@ async def calculate_spectrum(payload: Payload):
             databank=payload.database,
             use_cached=True,
         )
+        if simulate_slit is not None:
+            spectrum.apply_slit(payload.simulate_slit, "nm")
+        else:
+            pass
```
Comment (suzil) [inline server/main.py:65]: I would remove this "else pass" to keep the code more minimal so it's easier to read

**Comment by suzil** [comment]
I think the UI might be a bit confusing for users when they don't want to simulate a slit as right now they would have to leave the textfield with no value.

Screen Shot 2022-07-19 at 3 08 25 PM

I wonder if a better UI setup could be to use a "simulate slit" slider to ask the user whether or not they want to simulate a slit, and if the user activates the slider, then a new text field appears that prompts them for how big the slit should be.
 
Thoughts here @erwanp?

**Comment by arunavabasu-03** [comment]
@suzil, slider, or switch ??

**Comment by suzil** [comment]
Sorry I meant switch!

**Comment by erwanp** [comment]
Btw i'm noticing a small glitch 
image

**Comment by suzil** [comment]
@arunavabasu-03 In the future, would you write your commits according to this style guide? https://cbea.ms/git-commit/

Right now I'm trying to track when our frontend stopped running locally and some of the commit messages make it more difficult for me to go through the history and know what's going on because the commit messages are too vague.

File: frontend/src/components/fields/Mode.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ export const Mode: React.FC<DatabaseProps> = ({ control }) => {
             label="Select"
           >
             <MenuItem value={"absorbance"}>Absorbance</MenuItem>
-            <MenuItem value={"radiance_noslit"}>Radiance</MenuItem>
+            <MenuItem value={"radiance_noslit "}>Radiance </MenuItem>
```
Comment (suzil) [inline frontend/src/components/fields/Mode.tsx:34]:
```suggestion
            <MenuItem value={"radiance_noslit"}>Radiance </MenuItem>
```

File: frontend/src/components/CalcSpectrum.tsx [hunk 4]
```diff
@@ -35,9 +35,11 @@ export const CalcSpectrum: React.FC = () => {
     useState<CalcSpectrumPlotData | undefined>(undefined);
   const [isNonEquilibrium, setIsNonEquilibrium] = useState(false);
   const [useGesia, setUseGesia] = useState(false);
-
+  const [useSlit, setUseSlit] = useState(false);
+  const [useSlitSwitch, setuseSlitSwitch] = useState(false);
```
Comment (suzil) [inline frontend/src/components/CalcSpectrum.tsx:39]:
```suggestion
  const [useSlitSwitch, setUseSlitSwitch] = useState(false);
```

**Comment by arunavabasu-03** [comment]
@suzil  done

**Comment by suzil** [comment]
I want to run the backend locally to test the changes made to both the frontend and the backend, but I'm having this bug with GEISA: https://github.com/radis/radis/issues/497

File: frontend/src/components/types.ts [hunk 5]
```diff
@@ -14,5 +14,6 @@ export type FormValues = {
   trot?: number;
   pressure: number;
   path_length: number;
-  simulate_slit: boolean;
+  simulate_slit?: number;
+  useSimulateSlit: boolean;
```
Comment (suzil) [inline frontend/src/components/types.ts:18]: For consistency with the other form values, I would switch this to snake_case.

File: server/main.py [hunk 6]
```diff
@@ -60,14 +60,15 @@ async def calculate_spectrum(payload: Payload):
             databank=payload.database,
             use_cached=True,
         )
+        if simulate_slit is not None:
```
Comment (suzil) [inline server/main.py:63]: Since we're already passing the value, we could do something like `if useSimulateSlit:`

**Comment by suzil** [comment]
I would have the simulate slit amount appear below the switch so that the user filling out the form works in a single direction (down).

https://user-images.githubusercontent.com/13723264/181255968-41556b04-6c94-45d7-a2b1-08a607a03224.mov

**Review by suzil — CHANGES_REQUESTED** [review]
A couple of small comments but it looks great!

**Comment by erwanp** [comment]
Hello, couple suggestinos : 

- Use "Apply Instrumental Slit Function" for Switch
- As label of input box use "Slit Size"

Can you also make the Slit size input box appear on the right of the Switch ? , or, if it doesn't work well, below the switch ?

**Comment by arunavabasu-03** [comment]
@erwanp, check !!

Screenshot 2022-07-28 at 9 37 54 PM

**Review by suzil — APPROVED** [review]
Great work!

**Pull request merged**